### PR TITLE
FIX: GAT y_pred error when inconsistent axis

### PR DESCRIPTION
**Pull request by kingjr**
Some sklearn classifiers can predict y_pred_ with shape (n_trials, 1) instead of (n_trials,)

### CODE REVIEW

**Comment by kingjr** [comment]
see https://github.com/scikit-learn/scikit-learn/issues/5058

File: mne/decoding/tests/test_time_gen.py [hunk 1]
```diff
@@ -235,6 +236,12 @@ def test_generalization_across_time():
     gat.predict(epochs)
     assert_raises(ValueError, gat.predict, epochs[:10])
 
+    # Check that still works with classifier that output y_pred with
+    # shape = (n_trials, 1) instead of (n_trials,)
+    gat = GeneralizationAcrossTime(clf=RANSACRegressor(LinearRegression()))
+    gat.fit(epochs)
```
Comment (agramfort) [inline mne/decoding/tests/test_time_gen.py]: how long does it take to run? I would use as little number of epochs as possible.

File: mne/decoding/time_gen.py [hunk 2]
```diff
@@ -614,7 +614,7 @@ def _predict(X, estimators):
             predict_size = _y_pred.shape[1] if _y_pred.ndim > 1 else 1
             y_pred = np.ones((n_epochs, predict_size, n_clf))
         if predict_size == 1:
-            y_pred[:, 0, fold] = _y_pred
+            y_pred[:, 0, fold] = np.squeeze(_y_pred)
```
Comment (agramfort) [inline mne/decoding/time_gen.py]: squeeze is generally evil as a source of bugs in degenerate cases. Do:

```
if y_pred.ndim == 2 and y_pred.shape[1] == 1:
     y_pred = y_pred[:, 0]
```

Comment (kingjr) [inline mne/decoding/time_gen.py]: ok, well noted

File: mne/decoding/tests/test_time_gen.py (hunk 1 above)
Comment (kingjr) [inline mne/decoding/tests/test_time_gen.py]: now .053 s, I did cv=2, and classify one 1 time sample over 14 epochs.

**Comment by kingjr** [comment]
ready for merge if travis' happy

File: mne/decoding/tests/test_time_gen.py [hunk 3]
```diff
@@ -235,6 +235,15 @@ def test_generalization_across_time():
     gat.predict(epochs)
     assert_raises(ValueError, gat.predict, epochs[:10])
 
+    # Check that still works with classifier that output y_pred with
+    # shape = (n_trials, 1) instead of (n_trials,)
+    gat = GeneralizationAcrossTime(clf=RANSACRegressor(LinearRegression()),
+                                   cv=2)
+    epochs._data = epochs._data[:, :, :2]
+    epochs.times = epochs.times[:2]
```
Comment (agramfort) [inline mne/decoding/tests/test_time_gen.py]: that's not kosher. Use a proper:

```
epochs = epochs[:2]
epochs = epochs.crop(...)
```

**Comment by agramfort** [comment]
besides LGTM

**Comment by agramfort** [comment]
thanks @kingjr !
